### DISCUSSION TITLE
Multiple Search values of a topic data isn't working

### DISCUSSION THREAD

**Discussion by Tejeshwarraju** [tchiotludo/akhq#1079] [General]
Hi Team.

Running on akhq 0.20.0

Topic Data Lookup isn't working. Multi search criteria on a topic data is not working as expected. Search is performing an OR instead of AND.

Example: Trying to lookup for the messages having the values "CusId": "12345678" "action": "DISABLED" isn't returning intended data, instead returning all the messages.

This capabilities used to work and exist in older version of akhq. Is it bug on akhq v0.20.0 ?

would need some support here

-Regards,
Tej

**Comment by tchiotludo** [comment]
A fix must be done here : https://github.com/tchiotludo/akhq/commit/49cf8467ac3bdcd889382f17be7ae22f036aa4cc 
Can you try with dev version please ?

**Comment by makampf** [comment]
Is there already a solution to the problem? How can I search after strings containing double quotes and whitespaces?

**Reply by AlexisSouquiere** [reply]
This: `"\"value with whitespace\""` should work.
By enclosing with double quotes, AKHQ considers the string inside as a single pattern. If there are double quotes inside, you need to escape them with backslash

**Reply by makampf** [reply]
Thank you. I know this question is little off topic, but is it also necessary to escape other special characters like ' {' etc.? What is about newline?
(So in general, how can I search through JSON payloads?)

**Reply by AlexisSouquiere** [reply]
It's a possible (even if it's not really user-friendly). Suppose that you have this message in AKHQ
```json
{
  "field1": "value1",
  "field2": 1
}
```

You can search for the 2 fields using
`"\"field1\":\"value1\",\"field2\": 1"`

Meaning that you don't have line breaks or whitespace between the field name and the value. No need to escape the brackets for me. AKHQ pretty print the message but on the search you will have to do the following. Improvements should be done on this part but at least it's working right now
